### PR TITLE
ParticleEmitter Component

### PR DESCRIPTION
**Pull request by norda97**
Finishes Tasks:
 - Create Particle Emitter

Particle Emitter Component, will be expanded upon later.

### CODE REVIEW

File: LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "ECS/Component.h"
+#include "ECS/Entity.h"
+
+namespace LambdaEngine
+{
+
+
+	struct ParticleEmitterComponent
+	{
+		enum Shape
```
Comment (TheoBerlin) [inline LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h]: 1. I think you can move the declaration outside of the component
2. See the code standard for enums. They should either be prefixed by E or F.
3. `CONE` and `SPHERE` are both set to 0. Is there a particular reason for this? If not, is there a reason you want to explicitly set their values at all?
4. The enum can be named something more specific to its use case, like `EParticleShape`

File: LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+
+#include "ECS/Component.h"
+#include "ECS/Entity.h"
+
+namespace LambdaEngine
+{
+
+
```
Comment (TheoBerlin) [inline LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h]: Unnecessary whitespace

File: LambdaEngine/Source/Game/ECS/Components/Rendering/ParticleEmitter.cpp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#include "Game/ECS/Components/Rendering/ParticleEmitter.h"
+
+namespace LambdaEngine {
+
+
+}
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Game/ECS/Components/Rendering/ParticleEmitter.cpp]: Are you planning on using this file? Also, the first curly bracket is misplaced.

**Review by TheoBerlin — CHANGES_REQUESTED** [review]

File: LambdaEngine/Source/Game/ECS/Components/Rendering/ParticleEmitter.cpp (hunk 3 above)
Comment (norda97) [reply LambdaEngine/Source/Game/ECS/Components/Rendering/ParticleEmitter.cpp]: Not intended, will be removed

File: LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h (hunk 1 above)
Comment (norda97) [reply LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h]: Fixed, Thanks for tips!

File: LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h (hunk 2 above)
Comment (norda97) [reply LambdaEngine/Include/Game/ECS/Components/Rendering/ParticleEmitter.h]: Fixed

**Review by TheoBerlin — APPROVED** [review]
Looks good!

A note on the name 'emitter shape'; I think it could be improved a bit. Something like 'emitter pattern' might be more descriptive.
